app/controllers/overrides/devise_token_auth_overrides/registrations_controller.rb

```diff
@@ -5,6 +5,17 @@ class RegistrationsController < DeviseTokenAuth::RegistrationsController
       before_action :permit_extra_params, only: %i[create update]
       skip_before_action :validate_account_update_params, only: :update
 
+      def render_create_success
+        render json: {
+          status: 'success',
+          data: @resource.as_json.merge(type: @resource.type)
```

Comment:
Don't you think it'd be better to use the UserSerializer instead?
Here is how to get a hash from a fast_jsonapi serializer.
https://github.com/Youssefares/egsl-website-api/blob/master/app/services/paginated_serializable_service.rb#L15

Also I think overriding in each of the controllers is enough:
https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/controllers/devise_token_auth/application_controller.rb#L7

Reply:
if I use the UserSerializer the schema of the response will change, which is not expected by the front end. We can change it but we will have to notify the front end team. I went with the simpler solution of just adding the type field.

I don't understand the second comment. Do you mean "overriding the application controller is enough"?

Reply:
I think the frontend should adapt (cc: @TarekAlQaddy). The devise endpoints are the only ones still not conforming to jsonapi.org specs. For consistency I think it's better we use the serializer.

Reply:
Regarding the second point, just overriding the application controller won't work I think, because we still have to create controllers that would inherit from it. I am saying in the controllers you modified/created (which are all children of `DeviseTokenAuth::ApplicationController`), override that method instead of the success methods.

Reply:
I can adapt to using jsonapi.org specs in auth routes yes, Just to double check the only response I will change is the user data and it is gotten from `auth/login` and `auth/validate_token` routes?

Reply:
I think all routes that used to return a user object will change. 
- Create, Update, Delete User.
- Login, Validate token.